src/support/ios.c

```diff
@@ -1107,6 +1109,10 @@ int ios_getutf8(ios_t *s, uint32_t *pwc)
     c0 = (char)c;
     if ((unsigned char)c0 < 0x80) {
         *pwc = (uint32_t)(unsigned char)c0;
+        if (c == '\n')
+            s->u_colno = 0;
+        else
+            s->u_colno += utf8proc_charwidth(*pwc);
```

Comment:
I'd be a little worried about the performance of calling this on every character.

Reply:
I did a little benchmark parsing `s = "begin;\n" * "i = 1\n"^10000 * "end\n"` and `@time Meta.parse(s)`:

* Julia 0.6: 0.38 seconds
* Julia 0.7 master: 0.42 seconds
* this PR: 0.43 seconds

The difference is probably in the noise.